### DISCUSSION TITLE
What subQuery field is ?

### DISCUSSION THREAD

**Discussion by usluio** [sequelize/sequelize#17100] [Help & Questions] [answered]
subQuery: false || true

What is that?,
There is nothing about it in documentation. 

When I check in the library it looks like default is false, but if I do not include subQuery: false I got an error like this "Unknown column 'user_exam.startedAt' in 'order clause'"

Payload: 
const findAndCountAllOptions: FindAndCountOptions<StudentResultsAttributes> = {
    attributes: ['id', 'user_exam_id', 'num_of_correct', 'num_of_incorrect', 'num_of_empty', 'score', 'createdAt'],
    distinct: true,
    order: [[{ as: 'user_exam', model: models.users_exam }, 'startedAt', 'DESC']],
    offset: parseInt(input.offset),
    limit: input.limit ? parseInt(input.limit) : 5,
    include: [
      {
        as: 'user_exam',
        model: models.users_exam,
        where: {
          user_id: input.student_id
        } satisfies WhereOptions<UsersExamModel>,
        required: true,
        include: [
          {
            as: 'papper',
            model: models.pappers,
            attributes: ['title', 'id', 'typeOfPapper', 'tag_id', 'detailedReportSeeDate'],
            required: true,
            where: {
              typeOfPapper: input.paperType || { [Op.ne]: null },
              tag_id: input.tag_id || { [Op.ne]: null }
            } satisfies WhereOptions<PappersModel>,
            include: [
              {
                as: 'papper_booklet_juncs',
                model: models.papper_booklet_junc,
                include: [
                  {
                    attributes: ['id', 'title'],
                    as: 'booklet',
                    model: models.booklets
                  }
                ]
              }
            ]
          }
        ]
      }
    ],
    subQuery: false
  }

**Comment by ephys** [comment, marked as answer]
It's not documented because it was not designed to be used outside of our internals. We'll fix the code that makes using it a necessity but in the meantime we documented it in v7: https://sequelize.org/docs/v7/querying/select-in-depth/#referencing-associated-models-in-a-parent-where-clause
